src/main/java/gov/cms/madie/terminology/service/FhirTerminologyService.java

```diff
@@ -278,4 +277,104 @@ private void updateOrInsertAllCodeSystems(List<CodeSystem> codeSystemList) {
+              String oid = code.get("oid") != null ? code.get("oid").replaceAll("'|'", "") : null;
+
+              Optional<Map.Entry<String, String>> mappedVersion =
+                  mapToFhirVersion(code.get("version"), oid, codeSystemEntries);
+
+              if (mappedVersion.isPresent()) {
+                String vsacVersion = mappedVersion.get().getKey();
+                String fhirVersion = mappedVersion.get().getValue();
+
+                return retrieveCodes(
+                    codeName, codeSystemName, vsacVersion, fhirVersion, oid, apiKey);
+              }
+
+              return retrieveCodes(codeName, codeSystemName, null, null, oid, apiKey);
+            })
+        .collect(Collectors.toList());
+  }
+
+  private Optional<Map.Entry<String, String>> mapToFhirVersion(
+      String version, String oid, List<CodeSystemEntry> codeSystemEntries) {
+
+    if (oid == null) {
+      return Optional.empty();
+    }
+
+    Optional<Map.Entry<String, String>> result;
+    if (version == null) {
+      result =
+          codeSystemEntries.stream()
+              .filter(codeSystemEntry -> codeSystemEntry.getOid().toString().equals(oid))
+              .findFirst()
+              .map(
+                  codeSystemVersion ->
+                      Map.entry(
+                          codeSystemVersion.getVersions().get(0).getVsac(),
+                          codeSystemVersion.getVersions().get(0).getFhir()));
+    } else {
+      result =
+          codeSystemEntries.stream()
+              .filter(codeSystemEntry -> codeSystemEntry.getOid().toString().equals(oid))
+              .flatMap(codeSystemEntry -> codeSystemEntry.getVersions().stream())
+              .filter(codeSystemVersion -> codeSystemVersion.getVsac().equals(version))
+              .map(
+                  codeSystemVersion ->
+                      Map.entry(codeSystemVersion.getVsac(), codeSystemVersion.getFhir()))
+              .findFirst();
+    }
+
+    if (result.isPresent()) {
+      return result;
+    }
+    return result;
+  }
+
+  private Code retrieveCodes(
+      String codeName,
+      String codeSystemName,
+      String vsacVersion,
+      String fhirVersion,
+      String oid,
+      String apiKey) {
+    try {
+      if (StringUtils.isEmpty(codeName)
+          || StringUtils.isEmpty(codeSystemName)
+          || StringUtils.isEmpty(fhirVersion)) {
+        return null;
+      }
+      CodeSystem codeSystem =
+          codeSystemRepository.findByOidAndVersion(oid, fhirVersion).orElse(null);
+      if (codeSystem == null) {
+        return null;
+      }
+      String codeJson =
+          fhirTerminologyServiceWebClient.getCodeResource(codeName, codeSystem, apiKey);
+
+      Parameters parameters = fhirContext.newJsonParser().parseResource(Parameters.class, codeJson);
+      Code code =
+          Code.builder()
+              .name(codeName)
+              .codeSystem(codeSystemName)
+              .version(vsacVersion)
```

Comment:
Should pass `fhirVersion`. `vsacService.getCodeStatus` compares `version` against the fhirVersion in the mapping doc. 
```suggestion
              .version(fhirVersion)
```

Reply:
Turns out, we needed both versions, so I added an `svsVersion` entry to the DTO.